src/Compilers/CSharp/Test/Emit/Emit/NoPiaEmbedTypes.cs

```diff
@@ -2056,10 +2056,10 @@ public ITest28 Test()
 
 
             DiagnosticDescription[] expected = {
-                // (10,27): error CS1729: 'ITest28' does not contain a constructor that takes 1 arguments
+                // (10,20): error CS1729: 'ITest28' does not contain a constructor that takes 1 arguments
                 //         return new ITest28(1);
-                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "(1)").WithArguments("ITest28", "1")
-                                               };
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "ITest28").WithArguments("ITest28", "1").WithLocation(10, 20)
```

Comment:
:memo: We don't have access to the argument list anymore but now the diagnostic location is consistent with regular object creation - on the type itself. For `new()` we report it on the entire node.